sdk/loadtestservice/Azure.ResourceManager.LoadTestService/tests/ScenarioTests/LoadTestResourceOperations.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.LoadTestService.Models;
+using Azure.ResourceManager.LoadTestService.Tests.Helpers;
+using Azure.ResourceManager.Models;
+using Azure.ResourceManager.Resources;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.LoadTestService.Tests
+{
+    public class LoadTestResourceOperations : LoadTestServiceManagementTestBase
+    {
+        private LoadTestResourceCollection _loadTestResourceCollection { get; set; }
+        private LoadTestResource _loadTestResource { get; set; }
+        private LoadTestResourceData _loadTestResourceData { get; set; }
+
+        public LoadTestResourceOperations(bool isAsync) : base(isAsync) //, RecordedTestMode.Record)
+        {
+        }
+
+        [SetUp]
+        public async Task ClearAndInitialize()
+        {
+            if (Mode == RecordedTestMode.Record || Mode == RecordedTestMode.Playback)
+            {
+                await CreateCommonClient();
+            }
+            var resourceGroupName = Recording.GenerateAssetName("SdkRg");
+            await LoadTestResourceHelper.TryRegisterResourceGroupAsync(ResourceGroupsOperations, LoadTestResourceHelper.RESOURCE_LOCATION, resourceGroupName);
+
+            _loadTestResourceCollection = (await GetResourceGroupAsync(resourceGroupName)).GetLoadTestResources();
+            _loadTestResourceData = new LoadTestResourceData(LoadTestResourceHelper.RESOURCE_LOCATION);
+        }
+
+        [OneTimeTearDown]
+        public void Cleanup()
+        {
+            CleanupResourceGroups();
+        }
+
+        [RecordedTest]
+        [AsyncOnly]
+        public async Task LoadTestResourcesListBySubscription()
+        {
+            //// List LoadTest resources by subscription
+            AsyncPageable<LoadTestResource> loadtestResources = Subscription.GetLoadTestResourcesAsync();
+            await foreach (LoadTestResource loadtestResource in loadtestResources)
+            {
+                Assert.IsNotNull(loadtestResource.Data.Name);
+            }
+            Assert.IsTrue(await loadtestResources.GetAsyncEnumerator().MoveNextAsync());
+        }
+
+        [RecordedTest]
+        [AsyncOnly]
+        public async Task LoadTestResourceOperationTests()
+        {
+            var loadTestResourceName = Recording.GenerateAssetName("Sdk-LoadTestService-DotNet");
+
+            //// Create
+            ArmOperation<LoadTestResource> loadTestCreateResponse = await _loadTestResourceCollection.CreateOrUpdateAsync(WaitUntil.Completed, loadTestResourceName, _loadTestResourceData);
+            await loadTestCreateResponse.WaitForCompletionAsync();
```

Comment:
We will implicitly call `WaitForCompletionAsync` if you set the first parameter of lro to `WaitUntil.Completed`, therefore I think you can remove this line.